sdk/translation/ai-translation-text/samples-dev/breakSentence.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary This sample demonstrates how to make a simple call to the Azure Text Translator service to get sentences' boundaries.
+ */
+import TextTranslationClient, { ErrorResponseOutput, TranslatorCredential, InputTextItem, FindSentenceBoundariesQueryParamProperties, BreakSentenceItemOutput, isUnexpected } from "@azure-rest/ai-translation-text";
+
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const endpoint = process.env["ENDPOINT"] || "https://api.cognitive.microsofttranslator.com";
+const apiKey = process.env["TEXT_TRANSLATOR_API_KEY"] || "<api key>";
+const region = process.env["TEXT_TRANSLATOR_REGION"] || "<region>";
+
+export async function main() {
+  console.log("== Get Sentence Boundaries sample ==");
+
+  const translateCedential = new TranslatorCredential(apiKey, region);
+  const translationClient = TextTranslationClient(endpoint, translateCedential, undefined);
+
+  const inputText: InputTextItem[] = [
+    { text: "zhè shì gè cè shì。" }
+  ];
+  const parameters: FindSentenceBoundariesQueryParamProperties & Record<string, unknown> = {
+    language: "zh-Hans",
+    script: "Latn"
+  };
+  const breakSentenceResponse = await translationClient.path("/breaksentence").post({
+    body: inputText,
+    queryParameters: parameters
+  })
+
+  if (breakSentenceResponse.status !== "200") {
```

Comment:
I think `isUnexpected` already tests for this case, no?

Reply:
Yes, we will keep both checks.

Reply:
I am curious why both checks? @joheredi IIUC, `isUnexpected` makes sure to return true for 200 responses only, correct?

Reply:
I do prefer this check - one to check for http status code and other for unexpected body content. And I have seen this double check in tests in other projects.

Reply:
`isUnexpected` checks for the statusCode not the body content. So as you have it here it will never execute 

```ts
 if (isUnexpected(breakSentenceResponse)) {
    throw breakSentenceResponse.body;
  }
```

If you need to do something with the body of an unexpected response you need to do it here:

```ts
if (isUnexpected(breakSentenceResponse)) {
    // Already got an unexpected status code according to the TypeSpec definition
    // Do something with the failure body
    throw breakSentenceResponse.body;
  }
```

if you got an expected status code and want to check if the body is valid you need to do your own check after `isUnexpected`

```ts
if (isUnexpected(breakSentenceResponse)) {
    throw breakSentenceResponse.body;
  }

// Here you know the statuscode is 200 (or the union of other expected status codes)
if (breakSentenceResponse.body.foo !== "foo") {
   throw new Error("Unexpected value of foo");
}
```


isUnexpected gives you type narrowing which makes it easier for customers to work with the response type